torch/nn/functional.py

```diff
@@ -1569,6 +1569,25 @@ def sigmoid(input):
     return input.sigmoid()
 
 
+def hardsigmoid(input, inplace=False):
+    r"""hardsigmoid(input) -> Tensor
+
+    Applies the element-wise function :math:`\text{Hardsigmoid}(x) = \frac{ReLU6(x + 3)}{6}`
+
+    Args:
+        inplace: If set to ``True``, will do this operation in-place. Default: ``False``
+
+    See :class:`~torch.nn.Hardsigmoid` for more details.
+    """
+    # TODO: question for reviewers, is this JIT portion correct?
```

Comment:
Can you elaborate on why this special branch is needed?

Reply:
would love to know as well, all the other functions in this file have it so I made an educated guess that it's needed.  @raghuramank100 , any ideas on who might know?

Reply:
Oh when i look into it it's to not do the whole `__torch_function__` business if we're doing TorchScript: https://github.com/pytorch/pytorch/commit/fa80299bdf111ec3bc1ed1c4b3492bfe449817c7

cc @suo this is probably something that should be on our radar as something that TS doesn't support